main.tf

```diff
@@ -14,12 +14,18 @@
  * limitations under the License.
  */
 
+
+locals {
+  address = var.create_address ? join("", google_compute_global_address.default.*.address) : var.address
+  url_map = var.create_url_map ? join("", google_compute_url_map.default.*.self_link) : var.url_map
```

Comment:
FYI the join is a workaround from https://github.com/hashicorp/terraform/issues/16726#issuecomment-543122115 otherwise users will get errors whenever they try to import a resource into their tfstate (any resource, happened to me for GKE node pools)

```
Error: Invalid index

  on .terraform/modules/lb_http/main.tf line 22, in locals:
  22:   address = compact(concat([var.address], google_compute_global_address.default.*.address), )[0]
    |----------------
    | google_compute_global_address.default is empty tuple
    | var.address is null

The given key does not identify an element in this collection value.


Error: Invalid index

  on .terraform/modules/lb_http/main.tf line 23, in locals:
  23:   url_map = compact(concat([var.url_map], google_compute_url_map.default.*.self_link), )[0]
    |----------------
    | google_compute_url_map.default is empty tuple
    | var.url_map is null

The given key does not identify an element in this collection value.


Error: Invalid index

  on .terraform/modules/gke/modules/beta-private-cluster/main.tf line 37, in locals:
  37:   region   = var.region == null ? join("-", slice(split("-", var.zones[0]), 0, 2)) : var.region
    |----------------
    | var.zones is empty list of string

The given key does not identify an element in this collection value.
```